src/DependencyInjection/BaldinofRoadRunnerExtension.php

```diff
@@ -239,10 +269,178 @@ private function configureKv(array $config, ContainerBuilder $container): void
+                ->addArgument(new Reference(LoggerInterface::class))
+                ->addTag('monolog.logger', ['channel' => BaldinofRoadRunnerExtension::MONOLOG_CHANNEL])
+            ;
+
+            $defaultInterceptors[] = DoctrineORMInterceptor::class;
+        }
+
+        $container->setParameter('temporal.default_interceptors', $defaultInterceptors);
+
+        foreach ($config['clients'] as $name => $options) {
+            $container->register("temporal.client.{$name}.connection", Connection::class)
+                ->setFactory([ConnectionFactory::class, 'createFromArray'])
+                ->setArguments([
+                    '$options' => [
+                        'address' => $options['address'],
+                        'crt' => $options['client_key'] ?? null,
+                        'client_key' => $options['client_key'] ?? null,
+                        'client_pem' => $options['client_pem'] ?? null,
+                        'override_server_name' => $options['override_server_name'] ?? null,
+                    ],
+                ]);
+
+            $container->register("temporal.client.{$name}.service_client.factory", ServiceClientFactory::class)
+                ->setArguments([
+                    '$connection' => new Reference("temporal.client.{$name}.connection"),
+                ]);
+
+            $container->register("temporal.client.{$name}.service_client", ServiceClientInterface::class)
+                ->setFactory([new Reference("temporal.client.{$name}.service_client.factory"), '__invoke'])
+                ->setAutoconfigured(true)
+                ->setAutowired(true);
+
+            $container->register("temporal.client.{$name}.option", ClientOptions::class)
+                ->setFactory([ClientOptionsFactory::class, 'createFromArray'])
+                ->setArguments([
+                    '$options' => [
+                        'namespace' => $options['namespace'],
+                        'identity' => $options['identity'] ?? null,
+                        'query_reject_condition' => $options['query_reject_condition'] ?? null,
+                    ],
+                ]);
+
+            $container->register("temporal.client.{$name}.interceptors", SimplePipelineProvider::class)
+                ->addArgument(array_map(service(...), $options['interceptors'] ?? []));
+
+            $container->register("temporal.client.{$name}.factory", WorkflowClientFactory::class)
+                ->setArguments([
+                    '$serviceClient' => new Reference("temporal.client.{$name}.service_client"),
+                    '$dataConverter' => new Reference('temporal.data_converter'),
+                    '$clientOptions' => new Reference("temporal.client.{$name}.option"),
+                    '$interceptors' => new Reference("temporal.client.{$name}.interceptors"),
+                ]);
+
+            $container->register("temporal.client.{$name}", WorkflowClient::class)
```

Comment:
using the interface would be better 

```suggestion
            $container->register("temporal.client.{$name}", \Temporal\Client\WorkflowClientInterface::class)
```

also, since there's `temporal.client.{$name}.schedule` is used for `ScheduleClient`, it would be better to name this client as `temporal.client.{$name}.workflow`